### DISCUSSION TITLE
Usage of project and project-path?

### DISCUSSION THREAD

**Discussion by HachimanSec** [projectdiscovery/nuclei#4181] [Q&A] [answered]
Hi everyone, 

brief question about the -project and -project-path string options. How are they supposed to work? Is it supposed to create a subfolder and automatically store the output there? 

right now I use the -o option to specify the output file. 

Thanks!

**Comment by MetzinAround** [comment]
Hi @HachimanSec thanks for the question. 

I'm not entirely sure how defining a project using that flag works! I think it's different from output since it's in the optimization section of the flags, but the description seems to indicate it helps keep you from sending the same request multiple times somehow. 

Any insight @olearycrew ? I'm not familiar with that flag usage, and checking the docs didn't yield much for me.

**Reply by olearycrew** [reply, marked as answer]
I believe this is for saving the requests/responses over time so that on multiple scans of the same set of targets (a "project" in this context) you can avoid making the same request multiple times over multiple scans using nuclei.

```
[INF] Templates clustered: 4 (Reduced 2 Requests)
```

Nuclei automatically aggregates requests on a given scan and only makes the same request once (by default using `/tmp` to do this), but I think this flag is intended to save that information so you can get the same benefits of decreased requests over multiple runs of nuclei. If you save it to a known path, then instead of using `/tmp` nuclei will use the folder you specify, allowing this functionality to survive a single run of nuclei.

Hopefully that helps!
